src/xrpld/nodestore/detail/DatabaseRotatingImp.h

```diff
@@ -79,16 +79,21 @@ class DatabaseRotatingImp : public DatabaseRotating
     void
     sweep() override;
 
+    // Include the space in the name to ensure it can't be set in a file
+    static constexpr auto unitTestFlag = " unit_test";
+
 private:
     std::shared_ptr<Backend> writableBackend_;
     std::shared_ptr<Backend> archiveBackend_;
-    // This needs to be a recursive mutex because callbacks in `rotateWithLock`
-    // can call function that also lock the mutex. A current example of this is
-    // a callback from SHAMapStoreImp, which calls `clearCaches`. This
-    // `clearCaches` call eventually calls `fetchNodeObject` which tries to
-    // relock the mutex. It would be desirable to rewrite the code so the lock
-    // was not held during a callback.
-    mutable std::recursive_mutex mutex_;
+    bool const unitTest_;
+
+    // Implements the "UpgradeLockable" concept
+    // https://www.boost.org/doc/libs/1_86_0/doc/html/thread/synchronization.html#thread.synchronization.mutex_concepts.upgrade_lockable
```

Comment:
```suggestion
    // https://www.boost.org/doc/libs/1_83_0/doc/html/thread/synchronization.html#thread.synchronization.mutex_concepts.upgrade_lockable
```

We're currently using 1.83. Keeping it as-is is fine, but this is simply more accurate.